FlowCrypt/Controllers/Setup/SetupGenerateKeyViewController.swift

```diff
@@ -75,51 +77,114 @@ final class SetupGenerateKeyViewController: SetupCreatePassphraseAbstractViewCon
 
 extension SetupGenerateKeyViewController {
     private func setupAccountWithGeneratedKey(with passPhrase: String) {
-        Promise { [weak self] in
-            guard let self = self else { return }
-            self.showSpinner()
-
-            let userId = try self.getUserId()
-
-            try awaitPromise(self.validateAndConfirmNewPassPhraseOrReject(passPhrase: passPhrase))
-
-            let encryptedPrv = try self.core.generateKey(passphrase: passPhrase, variant: .curve25519, userIds: [userId])
-
-            try awaitPromise(self.backupService.backupToInbox(keys: [encryptedPrv.key], for: self.user))
-
-            let passPhrase = PassPhrase(value: passPhrase, fingerprints: encryptedPrv.key.fingerprints)
-
-            self.keyStorage.addKeys(keyDetails: [encryptedPrv.key], source: .generated, for: self.user.email)
-            self.passPhraseService.savePassPhrase(with: passPhrase, inStorage: self.shouldStorePassPhrase)
-
-            let updateKey = self.attester.updateKey(
-                email: userId.email,
-                pubkey: encryptedPrv.key.public,
-                token: self.storage.token
-            )
+        showSpinner()
+
+        cancellable = Just(passPhrase)
+            .setFailureType(to: Error.self)
+            .subscribe(on: DispatchQueue.global())
+            .flatMap(validateAndConfirm)
+            .flatMap(generateKey)
+            .flatMap(backupKey)
+            .flatMap(processKey)
```

Comment:
I know this is a common pattern in the iOS world but I don't like to tie results of one method to inputs of other methods THIS closely. Where you have to have a `ProcessKeyInput` returned by backupKey - that doesn't make any sense to me. Each method should be individual, inputs and outputs should not be tied to the exact chain that we happen to call it in.

Notice how the original code was written. I'd like it written with the same pattern, but using `Future` instead of `Promise`. I'm actively trying to avoid this tight chaining pattern showed here.

Look at this code:

```
            let userId = try self.getUserId()

            try awaitPromise(self.validateAndConfirmNewPassPhraseOrReject(passPhrase: passPhrase))

            let encryptedPrv = try self.core.generateKey(passphrase: passPhrase, variant: .curve25519, userIds: [userId])

            try awaitPromise(self.backupService.backupToInbox(keys: [encryptedPrv.key], for: self.user))

            let passPhrase = PassPhrase(value: passPhrase, fingerprints: encryptedPrv.key.fingerprints)

            self.keyStorage.addKeys(keyDetails: [encryptedPrv.key], source: .generated, for: self.user.email)
            self.passPhraseService.savePassPhrase(with: passPhrase, inStorage: self.shouldStorePassPhrase)

            let updateKey = self.attester.updateKey(
                email: userId.email,
                pubkey: encryptedPrv.key.public,
                token: self.storage.token
            )
```

It's very clear what inputs and outputs of each method is. I want to preserve that.

Reply:
In provided code you have a mix of sinc and async operations. This is the reason why `await` is used. Do you want to have the same pattern used in code with `Future`?

Reply:
Yes - I'd like to use a pattern that looks very similar to the original code, with minimal or no changes to the general structure of the code.

Reply:
OK, I reverted changed back and added wait method for Combine Publisher. But now I don't understand the goal of refactoring.

Reply:
:-) the goal is that we depracate the Google Promise library, and instead use a subset of Combine - just the `Future`. Right now, half the code is done with Promise and half with Future so we want to standardize on just writing async code with Future.

Does that make sense?

Reply:
Yes, now I understand what you need.